src/mono/mono/metadata/object.c

```diff
@@ -878,6 +878,11 @@ compute_class_bitmap (MonoClass *klass, gsize *bitmap, int size, int offset, int
 			pos += offset;
 
 			type = mono_type_get_underlying_type (field->type);
+
+			// If a struct has inline array attribute and type is not generic instance, consider it as an array
+			if (m_class_is_inlinearray (klass) && type->type != MONO_TYPE_GENERICINST && type->type != MONO_TYPE_TYPEDBYREF && type->type != MONO_TYPE_VALUETYPE)
+				type->type = MONO_TYPE_ARRAY;
```

Comment:
1. I don't think it's a great idea to modify type->type.  I think you're actually altering `field->type->type`.  That will probably break something unrelated.

2. Why are generic instances different?

Reply:
I agree. Would it make sense to do the following in case of inline array struct?
```c
bitmap [pos / BITMAP_EL_SIZE] |= ((gsize)1) << (pos % BITMAP_EL_SIZE);
*max_set = MAX (GINT_TO_UINT32(*max_set), pos);
```

If type is valuetype generic instance it falls through and invokes `compute_class_bitmap` recursively. By adding `type->type != MONO_TYPE_GENERICINST && type->type != MONO_TYPE_TYPEDBYREF && type->type != MONO_TYPE_VALUETYPE` it allows recursive call. 

If bitmap is calculated outside of switch statement, there is no need for type altering and generic instances condition.